src/OrchardCore/OrchardCore.Environment.Shell/ShellHost.cs

```diff
@@ -72,13 +72,23 @@ public void Initialize()
 
         public ShellContext GetOrCreateShellContext(ShellSettings settings)
         {
-            return _shellContexts.GetOrAdd(settings.Name, tenant =>
+            var shell = _shellContexts.GetOrAdd(settings.Name, tenant =>
             {
                 var shellContext = CreateShellContextAsync(settings).Result;
                 RegisterShell(shellContext);
 
                 return shellContext;
             });
+
+            if (shell.Released && shell.ActiveScopes == 0)
```

Comment:
I beleive that if it's `Released` we should just remove it, even if there are active scopes.